common/utils/include/claragenomics/utils/device_buffer.cuh

```diff
@@ -32,26 +31,24 @@ public:
 };
 
 template <typename T>
-class device_storage
+class device_buffer
 {
 public:
     using value_type = T;
-    device_storage() = delete;
-    device_storage(size_t n_elements, int32_t device_id)
+    device_buffer()  = delete;
+    device_buffer(size_t n_elements, int32_t device_id)
         : size_(n_elements)
-        , device_id_(device_id)
     {
-        CGA_CU_CHECK_ERR(cudaSetDevice(device_id_));
+        CGA_CU_CHECK_ERR(cudaSetDevice(device_id));
```

Comment:
Regarding ease of debugging... Something I experienced in my implementation and that will happen here as well is that only this line is reported, but not the line that actually wanted to make the allocation, meaning one has to set a breakpoint and check the stack trace in order to know where does the call actually comes from. I don't see an easy way to work around this this though...

Reply:
This something that concerns the implementation of CGA_CU_CHECK_ERR in general, not this class, right?

Reply:
Sure, but when `CGA_CU_CHECK_ERR` is used in the usual way (e.g. wrapping a `cudaMemcpy`) this is not an issue. Once a level of indirection is added (like in this case) this becomes an issue.
As said, I just wanted to point it out, I have no good idea how to fix it.
@akkamesh @tijyojwad

Reply:
I would use boost::stacktrace for this, but boost would be a pretty heavy dependency.